src/DelayedCommand.c

```diff
@@ -0,0 +1,253 @@
+
+    // Remove the command from the executable list
+    FRESULT fresult;
+    for (unsigned int i = 0; ; i++) {
+        QueuedCommand_t copy;
+        UINT length;
+
+        fresult = f_lseek(fp, sizeof(*command) * (i + 1));
+        if (fresult != FR_OK) {
+            goto error;
+        }
+
+        fresult = f_read(fp, &copy, sizeof(*command), &length);
+        if (fresult != FR_OK) {
+            goto error;
+        }
+
+        fresult = f_lseek(fp, sizeof(*command) * i);
+        if (fresult != FR_OK) {
+            goto error;
+        }
+
+        // We are done, EOF. The read-write head is now at the end of the last written
+        // command. Truncate the file here.
+        if (length != sizeof(*command)) {
+            fresult = f_truncate(fp);
+            if (fresult != FR_OK) {
+                goto error;
+            }
+
+            return;
+        }
+
+        fresult = f_write(fp, &copy, sizeof(*command), &length);
+        if (length != sizeof(*command)) {
+            ERR_LOG_ERROR("Unable to write to command file");
+            return;
+        }
+
+        if (fresult != FR_OK) {
+            goto error;
+        }
+    }
+
+error:
+    ERR_LOG_ERROR_F("Unable to remove executed command (%d)", fresult);
+}
+
+static bool findNextCommand(QueuedCommand_t *command) {
+    FRESULT fresult = f_rewind(fp);
+    if (fresult != FR_OK) {
+        goto error;
+    }
+
+    UINT length;
+    fresult = f_read(fp, command, sizeof(*command), &length);
+    if (fresult != FR_OK) {
+        goto error;
+    }
+
+    return length == sizeof(*command);
+
+error:
+    ERR_LOG_ERROR_F("Unable to read next command from file (%d)", fresult);
+    return false;
+}
+
+static bool shouldExecute(QueuedCommand_t *command) {
+    RTC_TimeTypeDef time;
+    RTC_DateTypeDef date;
+
+    if (EXT(CORE_getTime)(&time, &date, RTC_FORMAT_BCD) != HAL_OK) {
+        return false;
+    }
+
+#define DELAY_COMPARE(term) {           \
+    if (term != command->term) {        \
+        return term > command->term;    \
+    }                                   \
+}
+
+    DELAY_COMPARE(date.Year)
+    DELAY_COMPARE(date.Month)
+    DELAY_COMPARE(date.Date)
+    DELAY_COMPARE(time.Hours)
+    DELAY_COMPARE(time.Minutes)
+#undef DELAY_COMPARE
+
+    return time.Seconds >= command->time.Seconds;
+}
+
+static bool openFile(void) {
+    if (EXT(CORE_enterSDRegion)(osWaitForever) != osOK) {
+        return false;
+    }
+
+    FRESULT fresult = f_open(fp, DELAY_COMMAND_FILE, FA_READ | FA_WRITE | FA_OPEN_ALWAYS);
+    if (fresult != FR_OK) {
+        ERR_LOG_ERROR_F("Unable to open command file (%d)", fresult);
+        EXT(CORE_exitSDRegion)();
+        return false;
+    }
+
+    return true;
+}
+
+static void closeFile(void) {
+    FRESULT fresult = f_close(fp);
+    if (fresult != FR_OK) {
+        ERR_LOG_ERROR_F("Unable to close command file (%d)", fresult);
+    }
+
+    EXT(CORE_exitSDRegion)();
+}
+
+static void taskFunction(void const *arg) {
```

Comment:
Without calls to the task monitor this task will die.

Reply:
Yeah, I was not sure about this. My current working (untested) theory, is that if the task monitor is not told about a particular task then it won't monitor it.

Reply:
I'll leave this out until it is determined that we need it